org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/AdempiereWebUI.java

```diff
@@ -513,11 +513,13 @@ public void logout()
 	    	isAdminLogin  = (boolean)desktop.getSession().getAttribute(ISSOPrincipalService.SSO_ADMIN_LOGIN);
 	    
 	    boolean isSSOLogin = "Y".equals(Env.getContext(Env.getCtx(), Env.IS_SSO_LOGIN));
+		String provider = (String) desktop.getSession().getAttribute(ISSOPrincipalService.SSO_SELECTED_PROVIDER);
 	    String ssoLogoutURL = null;
-	    if (!isAdminLogin && isSSOLogin)
+	    if (!isAdminLogin && (isSSOLogin && !Util.isEmpty(provider)))
 	    {
-	    	ISSOPrincipalService service = SSOUtils.getSSOPrincipalService();
-	    	ssoLogoutURL = service.getLogoutURL();
+	    	ISSOPrincipalService service = SSOUtils.getSSOPrincipalService(provider);
+	    	if (service != null && Util.isEmpty(provider, true))
```

Comment:
isn't a null service a fatal exception here ? Also, looks strange to have the opposite provider is empty check here (opposite of the enclosing if block).

Reply:
@hengsin 
1. I think fatal exception should handle at login time, If service will be missing, it encounter NPE first on login. Agree?

2. Provider is empty when there is single SSO and we did not asked user to select on login screen.

Reply:
hi @dpansheriya ,

1. Agree but it is still good to be defensive here and no harm raising exception here if service is mising.
2. What I mean is the following 2 if contradict each other:
`if (!isAdminLogin && (isSSOLogin && !Util.isEmpty(provider)))`
`if (service != null && Util.isEmpty(provider, true))`

Reply:
This is fixed on latest commit. Note that when provider is not set then only we needs to redirect to SSO. otherwise we have to redirect user to login screen.